listenbrainz/webserver/static/js/src/user/Listens.tsx

```diff
@@ -612,6 +658,77 @@ export default class Listens extends React.Component<
     window.history.pushState(null, "", `?min_ts=${minTimestampInSeconds}`);
   };
 
+  getListenCard = (listen: Listen): JSX.Element => {
+    const { deletedListen } = this.state;
+    const { newAlert } = this.props;
+    const { currentUser } = this.context;
+    const isCurrentUser =
+      Boolean(listen.user_name) && listen.user_name === currentUser?.name;
+    const listenedAt = get(listen, "listened_at");
+    const recordingMSID = getRecordingMSID(listen);
+    const recordingMBID = getRecordingMBID(listen);
+    const artistMBIDs = getArtistMBIDs(listen);
+    const trackMBID = get(listen, "track_metadata.additional_info.track_mbid");
+    const releaseGroupMBID = getReleaseGroupMBID(listen);
+    const canDelete =
+      isCurrentUser && Boolean(listenedAt) && Boolean(recordingMSID);
+
+    const isListenReviewable =
+      Boolean(recordingMBID) ||
+      artistMBIDs?.length ||
+      Boolean(trackMBID) ||
+      Boolean(releaseGroupMBID);
+    // All listens in this array should have either an MSID or MBID or both,
+    // so we can assume we can pin them. Playing_now listens are displayed separately.
```

Comment:
This comment should probably be reviewed since playing_now listens are not separate anymore

Reply:
Done.